pkg/apis/servicecatalog/validation/instance.go

```diff
@@ -345,7 +412,138 @@ func ValidateServiceInstanceReferencesUpdate(new *sc.ServiceInstance, old *sc.Se
 	return allErrs
 }
 
+func validateObjectReferences(spec *sc.ServiceInstanceSpec, fldPath *field.Path) field.ErrorList {
+	var errMsg string
+	allErrs := field.ErrorList{}
+
+	if spec.ClusterServiceClassRef != nil && spec.ServiceClassRef != nil {
+		errMsg = "ClusterServiceClassRef and ServiceClassRef should never be set simultaneously"
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("clusterServiceClassRef"), errMsg))
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("serviceClassRef"), errMsg))
+	}
+
+	if spec.ClusterServicePlanRef != nil && spec.ServicePlanRef != nil {
+		errMsg = "ClusterServicePlanRef and ServicePlanRef should never be set simultaneously"
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("clusterServicePlanRef"), errMsg))
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("servicePlanRef"), errMsg))
+	}
+
+	return allErrs
```

Comment:
What do you think about adding checks that makes sure that when a cluster scoped class ref is set that a cluster scoped plan is also set in this function? Essentially make sure that we aren't mixing cluster scoped classes/plans with ns scoped classes and plans.

Reply:
EDIT: I found a test for that farther down below. I think we are good.